### PR TITLE
Updated mp endpoint

### PR DESCRIPTION
**Pull request by nivemaham**
This branch consists merges from `update_build` and `read_bin_csv` branches.
On top of that , i have implemented fixes such as
- Make rest-api -> management-portal requests with valid tokens
- Add client-scopes to client configs
- Efficiently store, load MP related resources in context. 
- Use actual instance of MP in integration test and remove wiremock
- Updated integration test to use a valid token
- Uses local MP-SNAPSHOT version with some fixes in OAuth-client-util
- Added project specific resources in separate endpoint class and test-cases in integration test. 
- Fixed chekstyle issues
- removed redundant config loading and removed unnecessary config properties and load from a single config file.

Note: I have commented out the e2e test, since i am working on a separate `docker-compose` file to have a reducedStack. I think it will be easier to review and will give clear separation of changes that are made. 

This is PR is rather large, since it contains two not merged branches.

### CODE REVIEW

**Comment by nivemaham** [comment]
Seems like, i didn't really fix all the checkstyle issues

File: gradle/docker.gradle
Comment (yatharthranjan) [inline gradle/docker.gradle:156]: If removed wiremock then i guess this block of code is not required.

Comment (nivemaham) [reply gradle/docker.gradle:156]: True. The wiremock is also used in the e2e test, which is not removed yet. I was working on another branch to fix the e2e test. This will be removed when we fully dockerize e2e test too. In this branch i am not using e2e tests

File: gradle/test.gradle
Comment (yatharthranjan) [inline gradle/test.gradle]: This can be removed as well

File: src/integrationTest/java/org/radarcns/integration/util/ApiClient.java
Comment (yatharthranjan) [inline src/integrationTest/java/org/radarcns/integration/util/ApiClient.java]: Can these can be static ?

Comment (yatharthranjan) [inline src/integrationTest/java/org/radarcns/integration/util/ApiClient.java]: Can these be static?

File: gradle/docker.gradle
Comment (yatharthranjan) [reply gradle/docker.gradle:156]: Ahh okay. got it! 👍

**Comment by nivemaham** [comment]
@yatharthranjan @blootsvoets Guys, can we make this a priority and finish reviewing, Please. If this is merged we can have designated branches to do upcoming changes. Since future changes could be large it is good have a separation.

File: Dockerfile
Comment (blootsvoets) [inline Dockerfile]: Use artifactory here. Preferably purge the localrepo dir from history.

File: bins.csv
Comment (blootsvoets) [inline bins.csv]: Move this file to the test resources directory. Preferably cut down the size as well.

File: build.gradle
Comment (blootsvoets) [inline build.gradle]: Remove this from the repo before merging.

File: src/integrationTest/images/wiremock/Dockerfile
Comment (blootsvoets) [inline src/integrationTest/images/wiremock/Dockerfile]: Is this not used anymore? Then it can be removed.

File: src/main/java/org/radarcns/listener/managementportal/PropertiesListener.java
Comment (blootsvoets) [inline src/main/java/org/radarcns/listener/managementportal/PropertiesListener.java]: Can be removed.

File: src/main/java/org/radarcns/security/filter/AuthenticationFilter.java
Comment (blootsvoets) [inline src/main/java/org/radarcns/security/filter/AuthenticationFilter.java]: Remove `Error` header. If needed at all, it should probably get a `RADAR-` prefix because it is not a standard HTTP header. Also, we should enumerate possible contents of the `Error` header.

File: src/main/java/org/radarcns/security/filter/AuthenticationFilter.java
Comment (blootsvoets) [inline src/main/java/org/radarcns/security/filter/AuthenticationFilter.java]: Likewise, this will not give a meaningful interaction, if the caller cannot anticipate on the possible responses in the JSON return object.

**Review by blootsvoets — COMMENTED** [review]
Some comments about the bins.csv implementation. I'll fix now.

File: src/integrationTest/images/wiremock/Dockerfile
Comment (nivemaham) [reply src/integrationTest/images/wiremock/Dockerfile]: Wiremock is not used in integration test. However, still used in e2e test which is currently not being checked. So effectively it is not used. I kept it till we come up with a working solution for e2e test which i was working on a separate branch.

File: build.gradle
Comment (nivemaham) [reply build.gradle]: We are depending on some changes from MP that are not released yet. We will have to wait for the next release of MP . I will enable snapshot releases of MP, then remove this

File: src/main/java/org/radarcns/listener/managementportal/ManagementPortalClient.java
Comment (blootsvoets) [inline src/main/java/org/radarcns/listener/managementportal/ManagementPortalClient.java]: Should throw an exception here.

File: src/main/java/org/radarcns/listener/managementportal/ManagementPortalClient.java
Comment (blootsvoets) [inline src/main/java/org/radarcns/listener/managementportal/ManagementPortalClient.java]: Close the response (try-with-resources)

**Review by blootsvoets — COMMENTED** [review]
ManagementPortalClient is not suitable for use yet. I will improve it before we can merge.

File: src/main/java/org/radarcns/listener/managementportal/ManagementPortalClient.java
Comment (nivemaham) [reply src/main/java/org/radarcns/listener/managementportal/ManagementPortalClient.java]: Indeed. There is still a lot of room for improvements. E.g. there is a local list of subjects which was used as cache. But it would introduce inconsistencies. I don't use them, but it is not completely removed. I think if we need a cache there, it should be more systematic.  Also i agree the exception handling can be improved.

File: src/integrationTest/java/org/radarcns/dao/AndroidDaoTest.java
Comment (yatharthranjan) [inline src/integrationTest/java/org/radarcns/dao/AndroidDaoTest.java]: Can the variable names also be `TIME_WINDOW` just for consistency

File: src/integrationTest/java/org/radarcns/integration/unit/ExpectedValueTest.java
Comment (yatharthranjan) [inline src/integrationTest/java/org/radarcns/integration/unit/ExpectedValueTest.java]: Same for all the tests

File: src/integrationTest/java/org/radarcns/integration/util/ExpectedDataSetFactory.java
Comment (yatharthranjan) [inline src/integrationTest/java/org/radarcns/integration/util/ExpectedDataSetFactory.java]: `timeWindow` here too

File: src/main/java/org/radarcns/security/filter/AuthenticationFilter.java
Comment (yatharthranjan) [reply src/main/java/org/radarcns/security/filter/AuthenticationFilter.java]: The client can rely on the status code to find out the problem. This is just for any client to debug the issue which resulted in the error status code since Dashboard is still under dev. Or would you suggest that an empty body in the response ?

File: src/main/java/org/radarcns/security/filter/AuthenticationFilter.java
Comment (nivemaham) [reply src/main/java/org/radarcns/security/filter/AuthenticationFilter.java]: Good point. Wondering whether ` res.setHeader("WWW-Authenticate", "Bearer");` would make any sense here?

**Review by blootsvoets — APPROVED** [review]
Fine to merge now, further development can be done in other branches.

**Comment by nivemaham** [comment]
:+1: Thanks both of you for your contributions. Long awaited merge.

**Pull request merged**
